src/compas_ags/rhino/formobject.py

```diff
@@ -148,7 +149,18 @@ def draw(self):
                         color[edge] = self.settings['color.compression']
 
             guids = self.artist.draw_edges(color=color)
-            self.guid_edge = zip(guids, edges)
+            if len(guids) == len(edges):
+                self.guid_edge = zip(guids, edges)
+            else:
+                drawn_edges = []
+                for edge in edges:
+                    u, v = edge
+                    pt1, pt2 = self.diagram.edge_coordinates(u, v)
+                    if geometric_key(pt1) == geometric_key(pt2):
```

Comment:
same comment as before. i would prefer a solution where indeed `None` values are added to the list instead of `GUID` when an edge is not drawn

Reply:
the property would then need to skip the zipped pair if the first item is `None`